ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1657,3 +1658,120 @@ uint8_t GCS_MAVLINK_Copter::high_latency_wind_direction() const
     return 0;
 }
 #endif // HAL_HIGH_LATENCY2_ENABLED
+
+// Send the mode with the given index (not mode number!) return the total number of modes
+// Index starts at 1
+uint8_t GCS_MAVLINK_Copter::send_available_mode(uint8_t index) const
+{
+    const Mode* modes[] {
+#if MODE_AUTO_ENABLED
+        &copter.mode_auto, // This auto is actually auto RTL!
+        &copter.mode_auto, // This one is really is auto!
```

Comment:
Why copy-paste twice?

Reply:
Once is for auto RTL mode the other is for auto mode. Auto RTL is not a full mode, just auto mode with a some hacks so it presents as its own mode.